integration-cli/check_test.go

```diff
@@ -74,25 +75,72 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func Test(t *testing.T) {
-	cli.SetTestEnvironment(testEnv)
-	fakestorage.SetTestEnvironment(&testEnv.Execution)
-	ienv.ProtectAll(t, &testEnv.Execution)
+func ensureTestEnvSetup(t *testing.T) {
+	testEnvOnce.Do(func() {
+		cli.SetTestEnvironment(testEnv)
+		fakestorage.SetTestEnvironment(&testEnv.Execution)
+		ienv.ProtectAll(t, &testEnv.Execution)
+	})
+}
+
+func TestDockerSuite(t *testing.T) {
```

Comment:
Looks like it still uses a top-level `TestxxSuite`, but at least it's now a unique name (instead of just `Test`); e.g. `TestDockerSuite/DockerSuite/TestAPIClientVersionOldNotSupported`

```
INFO: Testing against a local daemon
=== RUN   TestDockerSuite
=== RUN   TestDockerSuite/DockerSuite/TestAPIClientVersionOldNotSupported
=== RUN   TestDockerSuite/DockerSuite/TestAPICreateDeletePredefinedNetworks
```

Would it help if we changed this line to not include the testsuite name? https://github.com/moby/moby/blob/d32e6bbde815f37aa1cb23f602ab6371b509f3b2/internal/test/suite/suite.go#L39

Reply:
ugh, good catch.